src/vivarium_gates_nutrition_optimization/components/observers.py

```diff
@@ -144,6 +144,42 @@ def setup(self, builder: Builder) -> None:
             )
 
 
+class MaternalInterventionObserver:
+    configuration_defaults = {
+        "stratification": {
+            "maternal_interventions": {
+                "exclude": [],
+                "include": [],
+            }
+        }
+    }
+
+    def __repr__(self):
+        return "MaternalInterventionObserver()"
+
+    @property
+    def name(self):
+        return "maternal_intervention_observer"
+
+    #################
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.step_size = builder.time.step_size()
+        self.config = builder.configuration.stratification.maternal_interventions
+
+        for intervention in models.SUPPLEMENTATION_CATEGORIES:
+            builder.results.register_observation(
+                name=f"intervention_{intervention}_count",
+                pop_filter=f'alive == "alive" and intervention == "{intervention}" and tracked == True',
+                requires_columns=["alive", "intervention"],
```

Comment:
Are the required columns always just those that are in the pop filter above? (aside from the tracked column...and also how is that not required?)

Reply:
tracked (and only tracked) comes along by default in the init method of the `ResultsManager`. As for the others, you may also need to specify required columns / pipelines if you have an `aggregator` function, but in this case it is indeed just the columns from the filter.